File: base/src/main/java/com/windscribe/vpn/api/response/UserLoginResponse.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.api.response;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Mustafizur on 2017-09-08.
 */

@Keep
public class UserLoginResponse {

    @SerializedName("alc")
    @Expose
    private List<String> alcList;

    @SerializedName("billing_plan_id")
    @Expose
    private Integer billingPlanID;

    @SerializedName("email_status")
    @Expose
    private String emailStatus;

    @SerializedName("is_premium")
    @Expose
    private Integer isPremium;

    @SerializedName("loc_hash")
    @Expose
    private String locationHash;

    @SerializedName("loc_rev")
    @Expose
    private String locationRevision;

    @SerializedName("reg_date")
    @Expose
    private String registrationDate;

    @SerializedName("session_auth_hash")
    @Expose
    private String sessionAuthHash;

    @SerializedName("sip")
    @Expose
    private Sip sip;

    @SerializedName("traffic_max")
    @Expose
    private String trafficMax;

    @SerializedName("traffic_used")
    @Expose
    private String trafficUsed;

    @SerializedName("status")
    @Expose
    private Integer userAccountStatus;

    @SerializedName("email")
    @Expose
    private String userEmail;

    @SerializedName("user_id")
    @Expose
    private String userID;

    @SerializedName("username")
    @Expose
    private String userName;

    public List<String> getAlcList() {
        return alcList;
    }

    public Integer getBillingPlanID() {
        return billingPlanID;
    }

    public String getEmailStatus() {
        return emailStatus;
    }

    public Integer getIsPremium() {
        return isPremium;
    }

    public String getLocationHash() {
        return locationHash;
    }

    public String getLocationRevision() {
        return locationRevision;
    }

    public String getRegistrationDate() {
        return registrationDate;
    }

    public String getSessionAuthHash() {
        return sessionAuthHash;
    }

    public Sip getSip() {
        return sip;
    }

    public String getTrafficMax() {
        return trafficMax;
    }

    public String getTrafficUsed() {
        return trafficUsed;
    }

    public Integer getUserAccountStatus() {
        return userAccountStatus;
    }

    public String getUserEmail() {
        return userEmail;
    }

    public String getUserID() {
        return userID;
    }

    public String getUserName() {
        return userName != null ? userName : "na";
    }

    public boolean isInGhostMode() {
        return getUserName().equals("na");
    }

    @NonNull
    @Override
    public String toString() {
        return "UserLoginResponse{" +
                "session_auth_hash=" + sessionAuthHash +
                ", username='" + userName + '\'' +
                ", user_id='" + userID + '\'' +
                ", traffic_used='" + trafficUsed + '\'' +
                ", traffic_max='" + trafficMax + '\'' +
                ", status='" + userAccountStatus + '\'' +
                ", email='" + userEmail + '\'' +
                ", email_status='" + emailStatus + '\'' +
                ", billing_plan_id='" + billingPlanID + '\'' +
                ", is_premium='" + isPremium + '\'' +
                ", reg_date='" + registrationDate + '\'' +
                ", loc_rev='" + locationRevision + '\'' +
                ", loc_hash='" + locationHash + '\'' +
                '}';
    }
}

File: base/src/main/java/com/windscribe/vpn/api/response/UserLoginResponse.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.api.response

import androidx.annotation.Keep
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by Mustafizur on 2017-09-08.
 */
@Keep
class UserLoginResponse {

    @SerializedName("alc")
    @Expose
    val alcList: List<String>? = null

    @SerializedName("billing_plan_id")
    @Expose
    val billingPlanID: Int? = null

    @SerializedName("email_status")
    @Expose
    val emailStatus: String? = null

    @SerializedName("is_premium")
    @Expose
    val isPremium: Int? = null

    @SerializedName("loc_hash")
    @Expose
    val locationHash: String? = null

    @SerializedName("loc_rev")
    @Expose
    val locationRevision: String? = null

    @SerializedName("reg_date")
    @Expose
    val registrationDate: String? = null

    @SerializedName("session_auth_hash")
    @Expose
    val sessionAuthHash: String? = null

    @SerializedName("sip")
    @Expose
    val sip: Sip? = null

    @SerializedName("traffic_max")
    @Expose
    val trafficMax: String? = null

    @SerializedName("traffic_used")
    @Expose
    val trafficUsed: String? = null

    @SerializedName("status")
    @Expose
    val userAccountStatus: Int? = null

    @SerializedName("email")
    @Expose
    val userEmail: String? = null

    @SerializedName("user_id")
    @Expose
    val userID: String? = null

    @SerializedName("username")
    @Expose
    private val userNameValue: String? = null

    val userName: String
        get() = userNameValue ?: "na"

    val isInGhostMode: Boolean
        get() = userName == "na"

    override fun toString(): String {
        return "UserLoginResponse{" +
                "session_auth_hash=[REDACTED]" +
                ", username='" + userNameValue + '\'' +
                ", user_id='" + userID + '\'' +
                ", traffic_used='" + trafficUsed + '\'' +
                ", traffic_max='" + trafficMax + '\'' +
                ", status='" + userAccountStatus + '\'' +
                ", email='" + userEmail + '\'' +
                ", email_status='" + emailStatus + '\'' +
                ", billing_plan_id='" + billingPlanID + '\'' +
                ", is_premium='" + isPremium + '\'' +
                ", reg_date='" + registrationDate + '\'' +
                ", loc_rev='" + locationRevision + '\'' +
                ", loc_hash='" + locationHash + '\'' +
                '}'
    }
}
